server/src/main/java/io/spine/server/event/model/SubscriberMethod.java

```diff
@@ -66,10 +66,18 @@ protected SubscriberMethod(Method method, ParameterSpec<EventEnvelope> parameter
     @Override
     public DispatchKey key() {
         DispatchKey typeBasedKey = super.key();
+        return withFilter(typeBasedKey);
+    }
+
+    /**
+     * Applies {@link #filter() filter} if a specific one is supplied. Otherwise returns the
+     * supplied {@code key}.
+     */
+    DispatchKey withFilter(DispatchKey key) {
```

Comment:
To me `SubscriberMethod.withFilter(DispatchKey):DispatchKey` looks very strange in a package-level API.

Please consider rearranging things to hide this method.